TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/TrnRequestMetadataBackfillJob.cs

```diff
@@ -0,0 +1,358 @@
+                                    .FirstOrDefault(jObj => string.Equals(jObj.Value<string>("logicalName"), "dfeta_ninumber", StringComparison.OrdinalIgnoreCase));
+
+                                var applicationUser = audit.GetAttributeValue<EntityReference>("userid").Id;
+                                var createdOn = audit.GetAttributeValue<DateTime>("createdon");
+                                var firstName = firstNameObj?.Value<string>("newValue");
+                                var middleName = middleNameObj?.Value<string>("newValue");
+                                var lastName = lastNameObj?.Value<string>("newValue");
+                                var dob = dobObj?.Value<DateTime?>("newValue");
+                                var gender = genderObj?.Value<int?>("newValue");
+                                //var addressline1 = addressline1Obj.Value<string>("newValue");   //not in audit for creates
+                                //var addressline2 = addressline2Obj.Value<string>("newValue");   //not in audit for creates
+                                //var addressline3 = addressline3Obj.Value<string>("newValue");   //not in audit for creates
+                                //city                                                            //not in audit for creates
+                                //country                                                         //not in audit for creates
+                                var email = emailAddressObj?.Value<string>("newValue");
+                                var postcode = postalcodeObj?.Value<string>("newValue");
+                                var fullname = fullNameObj?.Value<string>("newValue");
+                                var nino = ninoObj?.Value<string>("newValue");
+
+                                //applicationUser & contact.dfeta_TrnRequestID
+                                var requestId = TrnRequestHelper.GetCrmTrnRequestId(applicationUser, contact.Value);
+
+                                recordsToInsert.Add(new TrnRequestMetadata
+                                {
+                                    ApplicationUserId = audit.GetAttributeValue<EntityReference>("userid").Id,
+                                    RequestId = requestId,
+                                    CreatedOn = audit.GetAttributeValue<DateTime>("createdon"),
+                                    DateOfBirth = dob.Value.ToDateOnlyWithDqtBstFix(isLocalTime: false),
+                                    EmailAddress = email ?? string.Empty,
+                                    IdentityVerified = false,
+                                    Name = new string[] { firstName, middleName, lastName }.Where(s => !string.IsNullOrEmpty(s)).ToArray(),
+                                    Gender = gender,
+                                    Postcode = postcode ?? string.Empty,
+                                    PotentialDuplicate = false,
```

Comment:
`PotentialDuplicate` should be `true` if a review task was created for the contact.